src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -540,25 +540,33 @@ namespace xclhwemhal2 {
       {
         //CR 966701: alignment to 4k (instead of mDeviceInfo.mDataAlignment)
         mDDRMemoryManager.push_back(new xclemulation::MemoryManager(it.size, it.base_addr, getpagesize(), it.tag));
+        std::string tag = it.tag;
+        int32_t index = it.index;
+
+        std::size_t found = tag.find("HOST");
+        if (found != std::string::npos) {
+          host_sptag_idx = index;
+        }
```

Comment:
Why all the copying to local variables?  Maybe simplify as:
```
if (it.tag.find("HOST") != std::string::npos)
  host_sptag_idx = it.index;
```

Reply:
Removed the local variables as suggested